src/librustc/session/config.rs

```diff
@@ -2350,6 +2348,15 @@ pub fn build_session_options_and_crate_config(
             }
         }
     };
+
+    // Disable hardware tracing when optimisations are enabled, switching to software tracing
+    // instead.
+    if opt_level != OptLevel::No {
+       cg.tracer = TracerMode::Software;
+    }
+
+    let cg = cg;
```

Comment:
what's the `let cg = cg` for?

Reply:
It makes `cg` immutable. It is original rustc code that I had to move down a bit in order to be able to mutate `cg.tracer`.

Reply:
Ah. Clever. Never seen that before.